### DISCUSSION TITLE
Is it possible to use Intents with the REST API?

### DISCUSSION THREAD

**Discussion by pranavbadami** [discordjs/discord.js#8668] [Q&A] [answered]
Hey folks, 

I think I'm missing something here, but I'm doing something like this in my code 
```
 const rest = new REST({
      version: "10",
    }).setToken(process.env["DISCORD_BOT_TOKEN"] ?? "");

    const members = (await rest.get(
      Routes.guildMembers(guildId),
      {}
    )) as APIGuildMember[];
```

However, this only returns a limited subset of members. I know this is a very common gotcha with the `GUILD_MEMBERS` intent when using the `Gateway`, but is it possible to send Intents along with REST API requests? 

Thanks in advance for the help!

**Comment by didinele** [comment, marked as answer]
Hiya! That REST endpoint does not provide all members in a guild regardless of your allowed intents. I believe it's paginated if my memory serves me right...?

If you want to get all members in a guild (i.e. how discord.js does it via the GuildMemberManager#fetch method), it's advised you use the gateway - there's a `REQUEST_GUILD_MEMBERS` payload that you can send, which indeed does require the intent if you desire to get *all* members.

The gateway then responds with `GUILD_MEMBER_CHUNK` packets.

**Reply by vladfrangu** [reply]
Adding to DD's comment, while some rest endpoints *are* affected by what intents you have enabled for your app in the Developer portal, you do not have to specify them! Enabling them is enough.
